metadata-db-app/src/cmr/metadata_db/services/concept_constraints.clj

```diff
@@ -94,41 +101,80 @@
+                        [(msg/concept-higher-transaction-id
+                          revision-id
+                          concept-id
+                          transaction-id
+                          this-revision-id
+                          this-transaction-id)])
                   (when (and (> revision-id this-revision-id)
                              (< transaction-id this-transaction-id))
-                        [(format (str "Revision [%d] of concept [%s] has transaction-id [%d] "
-                                      "which is lower than revision [%d] with transaction-id [%d]."
-                                     revision-id
-                                     concept-id
-                                     transaction-id
-                                     this-revision-id
-                                     this-transaction-id))]))))
+                        [(msg/concept-lower-transaction-id
+                          revision-id
+                          concept-id
+                          transaction-id
+                          this-revision-id
+                          this-transaction-id)]))))
           transaction-revisions)))
 
-;; Note - change back to a var once the enforce-granule-ur-constraint configuration is no longer
-;; needed. Using a function for now so that configuration can be changed in tests.
+(defn validate-pvn-equalities
+  "For any two given concepts, compares provider-ids, variable-names, and
+  native-ids between the two. Validation will return an error message if
+  provider-ids and variable-names match between the two, but the native-ids
+  don't.
+
+  Note that old-style validation functions (see `cmr.common.util`) expect
+  validation functions to return an empty list upon success and a list of
+  one or more error messages upon failure."
+  [new-concept old-concept]
+  (if (and (= (:provider-id new-concept)
+              (:provider-id old-concept))
+           (= (get-in new-concept [:extra-fields :variable-name])
+              (get-in old-concept [:extra-fields :variable-name]))
+           (not (= (:native-id new-concept)
+                   (:native-id old-concept))))
+    (msg/pvn-equality-failure new-concept old-concept)
+    []))
+
+(defn variable-pvn-constraint
+  "The 'pvn' constraint is for `provider-id`, `variable-name`, and `native-id`.
+  This function ensures that for these three attributes of a concept, updates
+  will only succeed if all three are the same for the most current revision-id.
+  If `provider-id` and `variable-name` are the same, but `native-id` is
+  different, this constraint will not be met and the valdiations will fail.
+
+  Any failures result in a list of error messages returned. Successful passing
+  of the constraint will result in an empty list being returned."
+  [db provider {:keys [native-id extra-fields] :as concept}]
+  (let [variable-name (:variable-name extra-fields)
+        concepts (find-latest-concepts
```

Comment:
`find-latest-matching-concepts` only handles one field; `find-latest-concepts` is the more general form needed here, since multiple fields are used ... what are you thinking?